code/script.js

```diff
@@ -2,334 +2,460 @@
-  // We also need a variable that stores the actual value of the question we've selected.
-
-  if (category === 'hair color') {
+  const chosenValue = questions.options[questions.selectedIndex].value
+  // category stores what option group (category) the question belongs to.
+  // chosenValue stores the actual value of the question we've selected.
+  if (category === 'hat') {
     currentQuestion = {
-      attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      attribute: 'hatColor',
+      value: chosenValue,
       category: category,
     }
-  } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
+  } else if (category === 'shirt') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: 'shirtColor',
+      value: chosenValue,
       category: category,
     }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+  } else if (category === 'other accessories') {
+    currentQuestion = {
+    attribute: chosenValue,
+    value: true, 
+    category: category,
+    }
+  } else if (category === 'facial hair') {
+    currentQuestion = {
+    attribute: chosenValue,
+    value: true, 
+    category: category,
+    }
+  } else {
+    currentQuestion = {
+    attribute: chosenValue,
+    value: true, 
+    category: category,
+    }
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
+/*This function should be invoked when you click on 'Find Out'.
+It compares currentQuestion with the values from the secret person. */
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+if (currentQuestion.attribute === 'hatColor') {
+  
+  if (currentQuestion.value === secret.hatColor) {
+      keep = true;
+  } else {
+    keep = false;
+  }
+} else if (currentQuestion.attribute === 'shirtColor') {
+  if (currentQuestion.value === secret.shirtColor) {
+    keep = true;
+  } else {
+    keep = false;
+  }
+} else if (currentQuestion.attribute === 'unibrow') {
+  if (currentQuestion.value === secret.unibrow) {
+    keep = true;
+  } else {
+    keep = false;
+  }
+} else if (currentQuestion.attribute === 'moustache') {
+  if (currentQuestion.value === secret.moustache) {
+    keep = true;
+  } else {
+    keep = false;
+  }
+} else if (currentQuestion.attribute === 'glasses') {
+  if (currentQuestion.value === secret.glasses) {
+    keep = true;
+  } else {
+    keep = false;
+  }
+} else {
+  if (currentQuestion.value === secret.scarf) {
+    keep = true;
+  } else {
+    keep = false;
+  }
+}
+group = currentQuestion.category;
```

Comment:
Are there a specific reason to declare **group** here? or can´t it be done directly in the **filterCharacthers** function?

Reply:
I think it needs to be declared so I can pass the arguments from output in checkQuestion to filterCharacters... but I'm not supersure